File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");

    int letters = 0;
    int words = 1;   // ilk kelimeyi hesaba katmak için
    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char c = text[i];

        // Harf kontrolü
        if (isalpha(c))
        {
            letters++;
        }

        // Kelime kontrolü
        else if (isspace(c))
        {
            words++;
        }

        // Cümle kontrolü
        else if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    // Ortalama hesap
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

def main():
    # Kullanıcıdan metni al
    text = get_string("Text: ")

    letters = 0
    words = 1  # İlk kelimeyi saymak için 1'den başlarız
    sentences = 0

    for char in text:
        # Harf sayımı
        if char.isalpha():
            letters += 1
        # Kelime sayımı (boşluklara göre)
        elif char == " ":
            words += 1
        # Cümle sayımı (. ! ?)
        elif char in [".", "!", "?"]:
            sentences += 1

    # L: 100 kelime başına düşen ortalama harf sayısı
    # S: 100 kelime başına düşen ortalama cümle sayısı
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # Coleman-Liau Formülü
    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)

    # Sonucu yazdır
    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")

if __name__ == "__main__":
    main()
